mlflow/store/sqlalchemy_store.py

```diff
@@ -206,43 +221,33 @@ def log_param(self, run_uuid, param):
             self._get_or_create(SqlParam, run_uuid=run_uuid, key=param.key,
                                 value=param.value)
         except sqlalchemy.exc.IntegrityError:
-            raise MlflowException('changing parameter {} value is not allowed'.format((run_uuid,
-                                                                                      param)),
+            self.session.rollback()
```

Comment:
Will this session actually relate to the get_or_create call? What would cause the session to be persistent?

Reply:
As I read it, it is an internal state management issue before calling DB again.